Matt/map fixes

mattbarker016 opened the pull request.

mattbarker016 committed:
route results fix, removed beta from feedback

mattbarker016 committed:
new bus asset, onboarding tweak

mattbarker016 committed:
better error analytics, other misc. fixes

mattbarker016 committed:
misc. network and analytic improvements

mattbarker016 committed:
Improved FavoritesTVC based on user feedback and design. Fixed shared route text for transfers

mattbarker016 committed:
live times on Route Detail view, needs some further testing when there's more live buses

mattbarker016 committed:
more live improvements

mattbarker016 committed:
banner debugging, added haptic feedback on route calculation load

mattbarker016 committed:
attempt to fix persistent banner

myo3 commented on TCAT/Controllers/FavoritesTableViewController.swift:
```diff
@@ -29,17 +30,32 @@ class FavoritesTableViewController: UITableViewController, UISearchBarDelegate {
-        resultsSection = Section(type: .searchResults, items: [ItemType]())
+        tableView.emptyDataSetSource = self
+        tableView.emptyDataSetDelegate = self
+        // tableView.reloadEmptyDataSet()
```
delete commented out code if uneeded

myo3 commented on TCAT/Controllers/FavoritesTableViewController.swift:
```diff
@@ -71,13 +87,12 @@ class FavoritesTableViewController: UITableViewController, UISearchBarDelegate {
-        let searchBar = UISearchBar()
         searchBar.isTranslucent = true
-        searchBar.placeholder = Constants.Phrases.searchPlaceholder
+        searchBar.placeholder = Constants.Phrases.favoritesPlaceholder
```
nice use of constants

myo3 commented on TCAT/Controllers/OnboardingViewController.swift:
```diff
@@ -73,7 +73,7 @@ class OnboardingViewController: PresentationController {
     let detailLabelMessages = [
         
-        "Beautiful and simple end-to-end navigation app for TCAT. Made by AppDev.",
+        "A b eautiful and simple end-to-end navigation app for TCAT. Made by AppDev.",
```
extra space

mattbarker016 replied:
thanks!

myo3 commented on TCAT/Controllers/RouteDetail+ContentViewController.swift:
```diff
@@ -27,9 +27,13 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
-    var networkTimer: Timer? = nil
-    /// Number of seconds to wait before auto-refreshing network call, timed with live indicator
-    var networkRefreshRate: Double = LiveIndicator.INTERVAL * 1.0
+    var liveTrackingNetworkTimer: Timer? = nil
```
no need for nil default value

myo3 commented on TCAT/Controllers/RouteDetail+ContentViewController.swift:
```diff
@@ -27,9 +27,13 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
+    /// Number of seconds to wait before auto-refreshing live tracking network call call, timed with live indicator
+    var liveTrackingNetworkRefreshRate: Double = LiveIndicator.INTERVAL * 1.0
+    
+    var busDelayNetworkTimer: Timer? = nil
```
same comment

myo3 commented on TCAT/Controllers/RouteDetail+ContentViewController.swift:
```diff
@@ -233,14 +232,14 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
     /// Show banner if no other status banner exists; turns status bar light
     func showBanner(_ message: String, status: BannerStyle) {
-        if self.banner == nil {
+        // if self.banner == nil {
```
delete commented out code

myo3 commented on TCAT/Controllers/RouteDetail+ContentViewController.swift:
```diff
@@ -173,23 +162,33 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
-        banner?.dismiss()
-        banner = nil
+        liveTrackingNetworkTimer?.invalidate()
+        liveTrackingNetworkTimer = nil
```
is it necessary to set it to nil? after you invalidate it its pretty much nil. just got to make sure before you use it again to set liveTrackingNetworkTimer to a new Timer

myo3 commented on TCAT/Controllers/RouteDetail+ContentViewController.swift:
```diff
@@ -173,23 +162,33 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
+        liveTrackingNetworkTimer?.invalidate()
+        liveTrackingNetworkTimer = nil
+        busDelayNetworkTimer?.invalidate()
+        busDelayNetworkTimer = nil
```
same comment

mattbarker016 committed:
PR changes, debark is gone, slight styling fixes

myo3 commented on TCAT/Controllers/RouteOptionsViewController.swift:
```diff
@@ -749,7 +758,7 @@ extension RouteOptionsViewController: UIViewControllerPreviewingDelegate {
         }
         
-        routeDetailViewController.preferredContentSize = CGSize(width: 0.0, height: 0.0)
+        routeDetailViewController.preferredContentSize = .zero
```
noice

myo3 commented on TCAT/Utilities/Extensions.swift:
```diff
@@ -175,7 +177,10 @@ extension JSON {
+        
+        // Zero out seconds, floor the time
+        let dateComponents = Calendar.current.dateComponents([.year, .month, .day, .hour, .minute], from: date)
+        return Calendar.current.date(from: dateComponents) ?? Date.distantPast
```
I created a `truncateSeconds(from date: Date)` in `Time.swift`. Its private but make it unprivate when you merge and use that for less repeated code

mattbarker016 committed:
Add Features, Testing required

Added live indicator to summaryView, added live times to all directions, removed shake to send feedback, general refactoring

mattbarker016 committed:
live indicator debugging, refactoring

mattbarker016 committed:
merged master

mattbarker016 committed:
print statements, build number

myo3 commented on TCAT/Views/SummaryView.swift:
```diff
@@ -34,6 +34,9 @@ class SummaryView: UIView {
     fileprivate var mainLabel = UILabel()
     
+    /// The live indicator
+    fileprivate var liveIndicator = LiveIndicator(size: .small, color: .clear)
```
why `fileprivate`? just make it `private` unless other classes in this SummaryView.swift file need to access these vars

mattbarker016 replied:
I thought Swift changed syntax to fileprivate. I should look up the difference

mattbarker016 replied:
TIL https://stackoverflow.com/questions/46588704/what-is-the-difference-between-private-and-fileprivate-in-swift-4

myo3 commented on TCAT/Controllers/RouteDetail+DrawerViewController.swift:
```diff
@@ -60,6 +64,24 @@ class RouteDetailDrawerViewController: UIViewController, UITableViewDataSource,
+        // Bus Delay Network Timer
+        if let timer = busDelayNetworkTimer {
+            timer.invalidate()
+        }
```
could just do the one-liner `busDelayNetworkTimer?.invalidate()`

myo3 commented on TCAT/Controllers/RouteDetail+DrawerViewController.swift:
```diff
@@ -60,6 +64,24 @@ class RouteDetailDrawerViewController: UIViewController, UITableViewDataSource,
+        }
+        busDelayNetworkTimer = Timer.scheduledTimer(timeInterval: busDelayNetworkRefreshRate, target: self, selector: #selector(getDelays),
+                                                    userInfo: nil, repeats: true)
+        busDelayNetworkTimer!.fire()
```
ik you set the timer just above but just to be safe use `?`

mattbarker016 committed:
last PR changes
